pkg/controller/provider/controller.go

```diff
@@ -196,11 +199,43 @@ func (r Reconciler) Reconcile(ctx context.Context, request reconcile.Request) (r
 			deployment)
 
 		// If the deployment does not exist
-		if k8serr.IsNotFound(err) {
-			r.CreateOVAServerDeployment(provider, ctx)
-		} else if err != nil {
+		if err != nil {
+			if k8serr.IsNotFound(err) {
+				err = r.CreateOVAServerDeployment(provider, ctx)
+				if err != nil {
+					r.handleServerCreationFailure(provider, err)
+					return
+				}
+				provider.Status.Phase = Staging
+				provider.Status.SetCondition(
+					libcnd.Condition{
+						Type:     Staging,
+						Status:   True,
+						Category: Required,
+						Message:  "The OVA server is being inizialized.",
+					})
+				err = r.Status().Update(context.TODO(), provider.DeepCopy())
+				if err != nil {
+					result.RequeueAfter = OvaReconcilerRetry
+					return
+				}
+				result.RequeueAfter = OvaReconcilerRetry
+				return
+			}
 			return
 		}
+
+		// The ova server pod is not running yet
+		if deployment.Status.AvailableReplicas == 0 {
+			if time.Since(provider.CreationTimestamp.Time) <= OvaTimeout {
```

Comment:
actually, does it work? I think we need something like `provider.CreationTimestamp.Add(OvaTimeout).Before(time.now())`

Reply:
I've tested it and it worked in the current implementation, but I can change if its not clear enough.

Reply:
ah ok, I see why - Duration is defined as `type Duration int64` so we get int64 on both sides
I prefer the fluent API I wrote above (didn't test that it even compiles though) but for your consideration

Reply:
changed to your suggestion + tested.